.github/workflows/main.yml

```diff
@@ -283,11 +300,18 @@ jobs:
         java: [ '17' ]
     steps:
 
+      - name: Check PR labels
+        if: ${{ github.event_name == 'pull_request' && join(github.event.pull_request.labels.*.name) == '' }}
+        run: |
+          echo "::error::PRs must be labeled, see: https://cwiki.apache.org/confluence/display/NETBEANS/PRs+and+You+-+A+reviewer+Guide"
+          exit 1
```

Comment:
this will fail when PRs are not labeled.

CI did always run some tests outside of the declared categories. The reasoning behind that was because I was worried about unlabeled PRs. Since this PR reduces the amount of testing outside of the declared labels, it will be enforced from now on (was discussed on apache slack with @neilcsmith-net ).

It should be easy to find labels even for trivial PRs, since we have enough labels which are generic and don't do anything. E.g https://github.com/apache/netbeans/labels/Editor, https://github.com/apache/netbeans/labels/CI or https://github.com/apache/netbeans/labels/Code%20cleanup. For readme edits we have https://github.com/apache/netbeans/labels/ci%3Ano-build

The IDE cluster job is always enabled, one day we should make it also configurable but lets do this incrementally.